src/core/utils/hooks/useOverflow.tsx

```diff
@@ -60,21 +69,26 @@ export const useOverflow = <T extends HTMLElement>(
     const availableWidth = containerRef.current.offsetWidth;
     const requiredWidth = containerRef.current.scrollWidth;
 
-    // hide items when there's no space available
-    if (availableWidth < requiredWidth && visibleCount > 1) {
-      setVisibleCount((count) => count - 1);
-      overflowBreakpoints.current.push(availableWidth);
-    }
-    // restore items when there's enough space again
-    else if (
-      overflowBreakpoints.current.length > 0 &&
-      availableWidth >
-        overflowBreakpoints.current[overflowBreakpoints.current.length - 1]
-    ) {
-      setVisibleCount((count) => count + 1);
-      overflowBreakpoints.current.pop();
+    if (availableWidth < requiredWidth) {
+      const avgItemWidth = requiredWidth / visibleCount;
```

Comment:
hmm..Interesting part. So initially, if there is more than 20 items, `visibleCount` would be 20, so avgItem is not really right one. Unless idea behind it is different?

Reply:
well ok, maybe kinda got it. looks smart, gj